src/app/components/Shots.tsx

```diff
@@ -0,0 +1,47 @@
+import React,{ useEffect } from "react";
+import { userInputsArr, ShotsTypes } from '@/store/inputStore';
+
+interface ShotsType{
+    examples: ShotsTypes[];
+}
+
+//shots are a single example
+function Shots({examples}:ShotsType){
```

Comment:
I don't think you need two nested functions, you can remove one of them
`const Shots = ({ examples }: ShotsType) => {
    if (!example.length > 0) {...}
    else ....
}`